### PR TITLE
kubelet pod-resources: add e2e for KubeletPodResourcesGet feature

### PR DESCRIPTION
**Pull request by moshe010**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Add node e2e test for KubeletPodResourcesGet feature
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: test/e2e_node/podresources_test.go [hunk 1]
```diff
@@ -568,6 +571,48 @@ func podresourcesGetAllocatableResourcesTests(ctx context.Context, cli kubeletpo
...
+	ginkgo.By("checking the output when only pods which don't require resources are present")
+	expected := []podDesc{
+		{
+			podName: "pod-00",
+			cntName: "cnt-00",
+		},
+	}
+	tpd.createPodsForTest(ctx, f, expected)
+	resp, err := cli.Get(ctx, &kubeletpodresourcesv1.GetPodResourcesRequest{PodName: "pod-00", PodNamespace: f.Namespace.Name})
+	framework.ExpectNoError(err)
```
Comment (ffromani) [inline test/e2e_node/podresources_test.go:590]: this is now recognized as antipattern. Please add an annotation describing the failure. Example (feel free to change as you see fit): 
```
framework.ExpectNoError(err, "Get() call failed for pod %s/%s", f.Namespace.Name, "pod-00")
```
The idea is the failures should clearly point the reason and help the troubleshooting. Just using `framework.ExpectNoError(err)` is too terse and unhelpful.  Please make sure the `ExpectNoError` you add here are all annotated.

We/I will do later sweeping changes to this file to fix the other occurrences.

Comment (ffromani) [reply test/e2e_node/podresources_test.go:590]: (sorry for forgetting to add the labels earlier)

Comment (moshe010) [reply test/e2e_node/podresources_test.go:590]: I split it to 2 commits one add the describing  for ExpectNoError  to the old code and second commit with my changes.

**Review by ffromani — COMMENTED** [review]
/lgtm

File: test/e2e_node/podresources_test.go [hunk 2]
```diff
@@ -568,6 +571,48 @@ func podresourcesGetAllocatableResourcesTests(ctx context.Context, cli kubeletpo
 	}
 }
 
+func podresourcesGetTests(ctx context.Context, f *framework.Framework, cli kubeletpodresourcesv1.PodResourcesListerClient) {
+	var err error
+	ginkgo.By("checking the output when no pods are present")
+	_, err = cli.Get(ctx, &kubeletpodresourcesv1.GetPodResourcesRequest{PodName: "test", PodNamespace: f.Namespace.Name})
+	framework.ExpectError(err, "pod not found")
```
Comment (ffromani) [inline test/e2e_node/podresources_test.go:578]: shouldn't we check we get an empty response?

File: test/e2e_node/podresources_test.go [hunk 3]
```diff
@@ -568,6 +571,48 @@ func podresourcesGetAllocatableResourcesTests(ctx context.Context, cli kubeletpo
...
+	framework.ExpectNoError(err, "matchPodDescWithResources() failed err %v", err)
+	tpd.deletePodsForTest(ctx, f)
+
+	tpd = newTestPodData()
+	ginkgo.By("checking the output when only pod require CPU")
+	expected = []podDesc{
+		{
+			podName:    "pod-01",
+			cntName:    "cnt-00",
+			cpuRequest: 2000,
```
Comment (ffromani) [inline test/e2e_node/podresources_test.go:603]: this is a burstable pod. Do you want also to check a guaranteed pod? (if so you need to set requests=limits, or just set limits and the system will automatically add requests=limits)

File: test/e2e_node/podresources_test.go [hunk 4]
```diff
@@ -838,6 +911,20 @@ var _ = SIGDescribe("POD Resources [Serial] [Feature:PodResources][NodeFeature:P
...
+				podName: "pod-01",
+				cntName: "cnt-01",
+			}
+			tpd := newTestPodData()
+			tpd.createPodsForTest(ctx, f, []podDesc{
+				desc,
+			})
+			expectPodResources(ctx, 1, cli, []podDesc{desc})
+
+			_, err = cli.Get(ctx, &kubeletpodresourcesv1.GetPodResourcesRequest{PodName: "pod-01", PodNamespace: f.Namespace.Name})
```
Comment (ffromani) [inline test/e2e_node/podresources_test.go:925]: do we want to check the value in the response here?

File: test/e2e_node/podresources_test.go (hunk 2 above)
Comment (moshe010) [reply test/e2e_node/podresources_test.go:578]: I will add

File: test/e2e_node/podresources_test.go (hunk 3 above)
Comment (moshe010) [reply test/e2e_node/podresources_test.go:603]: can you point me to example on how to check a guaranteed pod? where should I set the limits?

File: test/e2e_node/podresources_test.go (hunk 4 above)
Comment (moshe010) [reply test/e2e_node/podresources_test.go:925]: will add

File: test/e2e_node/podresources_test.go (hunk 3 above)
Comment (moshe010) [reply test/e2e_node/podresources_test.go:608]: see it already been set here https://github.com/kubernetes/kubernetes/blob/master/test/e2e_node/podresources_test.go#L99-L100.
So it seem it doing already what you want. Isn't it?

Comment (ffromani) [reply test/e2e_node/podresources_test.go:608]: yes, right, we should be good.

**Comment by klueska** [comment]
I am going to preliminarily approve this (with a hold).
I will let @ffromani do another pass and give his LGTM 
Feel free to unhold it once he has signed off.

/approve
/hold

File: test/e2e_node/podresources_test.go [hunk 5]
```diff
@@ -568,6 +571,53 @@ func podresourcesGetAllocatableResourcesTests(ctx context.Context, cli kubeletpo
...
 
+func podresourcesGetTests(ctx context.Context, f *framework.Framework, cli kubeletpodresourcesv1.PodResourcesListerClient) {
+	//var err error
+	ginkgo.By("checking the output when no pods are present")
+	expected := []podDesc{}
+	resp, err := cli.Get(ctx, &kubeletpodresourcesv1.GetPodResourcesRequest{PodName: "test", PodNamespace: f.Namespace.Name})
+	podResourceList := []*kubeletpodresourcesv1.PodResources{resp.GetPodResources()}
+	framework.ExpectError(err, "pod not found")
+	res := convertToMap(podResourceList)
+	err = matchPodDescWithResources(expected, res)
```
Comment (ffromani) [inline test/e2e_node/podresources_test.go:582]: ok this is a bit convoluted but works. If needed, we can simplify later.

**Comment by ffromani** [comment]
/retest

the unit test failure looks like a flake

**Comment by moshe010** [comment]
@ffromani do you think it ready for merge?

**Comment by ffromani** [comment]
/lgtm

This PR seems ok and we can extend coverage later

**Pull request merged**